filesystems/cvmfs/fs.go

```diff
@@ -0,0 +1,58 @@
+// +build linux
+
+package main
+
+import (
+	"fmt"
+	"strings"
+	"path/filepath"
+	"os"
+	"syscall"
+
+	"github.com/containerd/containerd/plugin"
+	fsplugin "github.com/ktock/remote-snapshotter/filesystems"
+)
+
+func init() {
+	plugin.Register(&plugin.Registration{
+		Type:   fsplugin.RemoteFileSystemPlugin,
+		Config: &Config{},
+		ID:     "cvmfs",
+		InitFn: func(ic *plugin.InitContext) (interface{}, error) {
+			config, ok := ic.Config.(*Config)
+			if !ok {
+				return nil, fmt.Errorf("invalid stargz configuration")
+			}
+			ic.Meta.Exports["root"] = ic.Root
+
+			return NewFilesystem(ic.Root, config)
+		},
+	})
+}
+
+type filesystem struct {}
+
+type Config struct {}
+
+func NewFilesystem(root string, config *Config) (*filesystem, error) {
+	return &filesystem{}, nil
+}
+
+func (fs *filesystem) Mount(ref, digest, moutpoint string) error {
+	digest = strings.Split(digest, ":")[1]
+	firstTwo := digest[0:2]
+	path := filepath.Join("/", "cvmfs", "unpacked.cern.ch", ".layer", firstTwo, digest)
+	if _, err := os.Stat(path); os.IsNotExist(err) {
+		return fmt.Errorf("layer %s not in the cvmfs repository", digest)
+	}
+	err := syscall.Mount(path, moutpoint, "", syscall.MS_BIND, "")
+	if err != nil { 
+		fmt.Printf("%+v\n", err)
+	}
+	return err
+}
+
+func (fs *filesystem) Check(moutpoint string) error {
+	fmt.Printf("mountpoint: %+v\n", moutpoint)
+	return fmt.Errorf("We did nothing, we are not really sure about this yet!")
+}
```

Comment:
Sorry for this unclear API function. This intends to be used for checking the existence of the layer by containerd. Remote snapshots can be unavailable during runtime (e.g. because of NW failure, etc.) so for resiliency purpose, this function will be called every time this layer is used by containerd. I'm planning to add error handling and fallback to containerd core. Please see the discussion in containerd community: https://github.com/containerd/containerd/issues/3731#issuecomment-539883710

Reply:
I updated remote snapshotter to leverage the `Check` API to make sure all snapshots are available(Commit: 3cc367b79ea0cd28a069f9fadfc525b246fe9d14). I will post another PR on containerd to handle the case of snapshots unavailable.

Reply:
Ahahaha nono it is not the API unclear, it was me that I didn't need it!

I don't want this to be production code, but it just need to be somewhere so that people interested in it can have a look.